File: app/src/main/java/com/example/firabasefirstexperience/java/activity/StartLoadingActivity.java
package com.example.firabasefirstexperience.java.activity;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.os.Handler;

import com.example.firabasefirstexperience.kotlin.StartActivity;
import com.google.firebase.auth.FirebaseAuth;
import com.google.firebase.auth.FirebaseUser;

public class StartLoadingActivity extends AppCompatActivity {

    private FirebaseUser fbUser;

    private static final int LOADING_SCREEN_SHOW_TIME = 2000;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        fbUser = FirebaseAuth.getInstance().getCurrentUser();

        new Handler().postDelayed(new Runnable() {
            @Override
            public void run() {
                if (fbUser != null) {
                    Intent intent = new Intent(StartLoadingActivity.this, MainActivity.class);
                    startActivity(intent);
                    finish();
                } else {
                    Intent intent = new Intent(StartLoadingActivity.this, StartActivity.class);
                    startActivity(intent);
                    finish();
                }
            }
        }, LOADING_SCREEN_SHOW_TIME);
    }
}

File: app/src/main/java/com/example/firabasefirstexperience/kotlin/activity/StartLoadingActivity.kt
package com.example.firabasefirstexperience.kotlin.activity

import android.content.Intent
import android.os.Bundle
import android.os.Handler
import androidx.appcompat.app.AppCompatActivity
import com.example.firabasefirstexperience.R
import com.google.firebase.auth.FirebaseAuth

class StartLoadingActivity : AppCompatActivity() {

    private val LOADING_SCREEN_SHOW_TIME = 2000

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_start_loading)

        val user = FirebaseAuth.getInstance().currentUser

        Handler().postDelayed({
            if (user != null) {
                val intent = Intent(this@StartLoadingActivity, MainActivity::class.java)
                startActivity(intent)
                finish()
            } else {
                val intent = Intent(this@StartLoadingActivity, StartActivity::class.java)
                startActivity(intent)
                finish()
            }
        }, LOADING_SCREEN_SHOW_TIME.toLong())
    }
}